### PR TITLE
Add mps tests to CI

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
Adds tests on MPS device after the M1 runners introduction
https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

#### Type of change
- [x] 🧪 Tests Cases

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]
LGTM. And amazing outcome ! waiting for @johnnv1 approval

**Comment by johnnv1** [comment]
i think we need to update the kornia-rs build to have m1 compatibility, otherwise, it will keep falling on CI -- https://github.com/kornia/kornia/actions/runs/7741816134/job/21109568889?pr=2775#step:2:165

**Comment by johnnv1** [comment]
@ducha-aiki for now, remove the version pinned on requirements-dev -- I think we should have some version compatible with m1 (maybe it's 0.0.7) -- https://github.com/kornia/kornia/blob/d5052d1d00f910568dc82b5be03b9648960f590c/requirements/requirements-dev.txt#L2

File: .github/workflows/scheduled_test_cpu.yml
Comment (johnnv1) [inline .github/workflows/scheduled_test_cpu.yml:60]:
```suggestion
```

**Comment by johnnv1** [comment]
@ducha-aiki now it's running 

some actual numbers
- 8845 failed
- 878 passed

**Comment by ducha-aiki** [comment]
Given the error, it seems that MPS is somehow disabled in that runner. So no sense in running tests there yet.

```
 RuntimeError: MPS backend out of memory (MPS allocated: 0 bytes, other allocations: 0 bytes, max allowed: 7.93 GB). Tried to allocate 2.50 KB on private pool. Use PYTORCH_MPS_HIGH_WATERMARK_RATIO=0.0 to disable upper limit for memory allocations (may cause system failure).
```

**Pull request closed without merging**
